src/Take.Blip.Builder/Utils/SalesForce/SalesForceClient.cs

```diff
@@ -0,0 +1,156 @@
+                {SalesForceConstants.REFRESH_TOKEN, salesForceConfig.RefreshToken }
+            };
+
+            var responseToken = string.Empty;
+
+            try
+            {
+                using (HttpClient client = new HttpClient())
+                {
+                    HttpResponseMessage response = client.PostAsync(
+                        $"{_configuration.SalesForceBaseUri}{SalesForceUriTemplates.REFRESH_TOKEN}",
+                        new FormUrlEncodedContent(body)
+                        ).Result;
+                    responseToken = await response.Content.ReadAsStringAsync();
+                };
+                return JsonConvert.DeserializeObject<AuthorizationResponse>(responseToken);
+            }
+            catch (Exception ex)
+            {
+                _logger.Error(ex, $"An exception occurred while processing a Sales Force HTTP request");
+                throw;
+            }
+        }
+
+        public async Task<LeadResponse> CreateLeadAsync(CrmSettings registerLeadSettings, AuthorizationResponse authorization)
+        {
+            var response = string.Empty;
+            var statusCode = HttpStatusCode.OK;
+
+            try
+            {
+                using (HttpClient client = new HttpClient())
+                {
+                    client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(
+                        authorization.TokenType,
+                        authorization.AccessToken
+                        );
+                    HttpResponseMessage responseMessage = client.PostAsync(
+                        $"{authorization.InstanceUrl}{SalesForceUriTemplates.CREATE_LEAD}",
+                        new StringContent(
+                            registerLeadSettings.LeadBody as string,
+                            Encoding.UTF8,
+                            "application/json"
+                            )
+                        ).Result;
+                    statusCode = responseMessage.StatusCode;
+                    response = await responseMessage.Content.ReadAsStringAsync();
+                };
+                if (statusCode == HttpStatusCode.BadRequest)
+                {
+                    var convertedErrorResponse = JsonConvert.DeserializeObject<List<LeadResponse>>(response);
+                    return convertedErrorResponse.First();
+                }
+
+                return JsonConvert.DeserializeObject<LeadResponse>(response);
+            }
+            catch (Exception ex)
+            {
+                _logger.Error(ex, $"An exception occurred while processing a Sales Force HTTP request");
+                throw;
+            }
+        }
+
+        public async Task<JObject> GetLeadAsync(CrmSettings settings, AuthorizationResponse authorization, CancellationToken cancellationToken)
+        {
+            var response = string.Empty;
+            var statusCode = HttpStatusCode.OK;
```

Comment:
Since this will not be used outside the try you can declare it when calling PostAsync/ReadAsStringAsync.